client/src/pages/recordings-browser/AnnotationModal.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 Microsoft Corporation
+// Copyright (c) 2023 Marc Lichtman
+// Licensed under the MIT License
+
+import Plot from 'react-plotly.js';
+import React, { useEffect, useState } from 'react';
+import { Annotation } from '@/utils/sigmfMetadata';
+import { template } from '@/utils/plotlyTemplate';
+
+interface Props {
+  setShowModal: any;
+  fileName: String;
+  annotations: Annotation[];
+}
+
+export const AnnotationModal = ({ setShowModal, fileName, annotations }: Props) => {
+  const [pieChartValues, setPieChartValues] = useState([]); // so that it can be directly used in the pie chart https://plotly.com/javascript/pie-charts/
+  const [pieChartLabels, setPieChartLabels] = useState([]);
+
+  // Calc Pie chart data
+  useEffect(() => {
+    const counts = {};
+    annotations.forEach((item) => {
+      const label = item.getLabel();
+      if (label in counts) {
+        counts[label] = counts[label] + 1;
+      } else {
+        counts[label] = 1;
+      }
+    });
+    const vals = [];
+    const labels = [];
+    Object.entries(counts).forEach(([k, v]) => {
+      labels.push(k);
+      vals.push(v);
+    });
+    setPieChartValues(vals);
+    setPieChartLabels(labels);
+  }, [annotations]); // TODO make sure this isnt going to be sluggish when currentSamples is huge
+
+  const annotationsData = annotations?.map((item, index) => {
+    const deepItemCopy = JSON.parse(JSON.stringify(item));
+    // remove the main feilds so that everything left is shown in "other"
+    delete deepItemCopy['core:sample_start'];
+    delete deepItemCopy['core:sample_count'];
+    delete deepItemCopy['core:freq_lower_edge'];
+    delete deepItemCopy['core:freq_upper_edge'];
+    delete deepItemCopy['core:label'];
+    delete deepItemCopy['core:description'];
+    return (
+      <tr key={index} className="h-12">
+        <td>{item['core:sample_start']}</td>
```

Comment:
it might be worth adding these to the sigmfmetadata file so we have .getSampleStart .getSampleCount etc